api/rbac/rbac.go

```diff
@@ -0,0 +1,191 @@
+	}
+
+	return enforcer, refreshEnforcerInBackground(ctx, kubeClient, enforcer, l)
+}
+
+// UserGetter is a function that extracts the user from the JWT token.
+func UserGetter(c echo.Context) (string, error) {
+	token, ok := c.Get("user").(*jwt.Token) // by default token is stored under `user` key
+	if !ok {
+		return "", errors.New("failed to get token from context")
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims) // by default claims is of type `jwt.MapClaims`
+	if !ok {
+		return "", errors.New("failed to get claims from token")
+	}
+
+	subject, err := claims.GetSubject()
+	if err != nil {
+		return "", errors.Join(err, errors.New("failed to get subject from claims"))
+	}
+
+	issuer, err := claims.GetIssuer()
+	if err != nil {
+		return "", errors.Join(err, errors.New("failed to get issuer from claims"))
+	}
+
+	if issuer == session.SessionManagerClaimsIssuer {
+		return strings.Split(subject, ":")[0], nil
+	}
+	return subject, nil
+}
+
+// NewEnforceHandler returns a function that checks if a user is allowed to access a resource.
+func NewEnforceHandler(basePath string, enforcer *casbin.Enforcer) func(c echo.Context, user string) (bool, error) {
+	return func(c echo.Context, user string) (bool, error) {
+		pathResourceMap := map[string]string{
+			basePath + "/backup-storages":                                           "backup-storages",
+			basePath + "/backup-storages/:name":                                     "backup-storages",
+			basePath + "/monitoring-instances":                                      "monitoring-instances",
+			basePath + "/monitoring-instances/:name":                                "monitoring-instances",
+			basePath + "/namespaces/:namespace/database-engines":                    "database-engines",
+			basePath + "/namespaces/:namespace/database-engines/:name":              "database-engines",
+			basePath + "/namespaces/:namespace/database-clusters":                   "database-clusters",
+			basePath + "/namespaces/:namespace/database-clusters/:name":             "database-clusters",
+			basePath + "/namespaces/:namespace/database-clusters/:name/backups":     "database-clusters",
+			basePath + "/namespaces/:namespace/database-clusters/:name/credentials": "database-clusters",
+			basePath + "/namespaces/:namespace/database-clusters/:name/pitr":        "database-clusters",
+			basePath + "/namespaces/:namespace/database-clusters/:name/restores":    "database-clusters",
+			basePath + "/namespaces/:namespace/database-clusters/:name/components":  "database-clusters",
+			basePath + "/namespaces/:namespace/database-cluster-backups":            "database-cluster-backups",
+			basePath + "/namespaces/:namespace/database-cluster-backups/:name":      "database-cluster-backups",
+			basePath + "/namespaces/:namespace/database-cluster-restores":           "database-cluster-restores",
+			basePath + "/namespaces/:namespace/database-cluster-restores/:name":     "database-cluster-restores",
```

Comment:
@percona/everest-backend-devs What do you guys think about automating this part, so that we don't have to manually update this code everytime a new path is added?

For example, what I had in mind is - read the swagger spec, parse the URL and extract the resource name out of it and build this map

Reply:
yes, would be great

Reply:
OK, will do in a follow up PR. I feel this PR is already too big and complex already..